apps/stencil_chain/stencil_chain_generator.cpp

```diff
@@ -50,23 +50,37 @@ class StencilChain : public Halide::Generator<StencilChain> {
         } else if (get_target().has_gpu_feature()) {
             // GPU schedule
 
-            // 5.90 ms on a 2060 RTX
+            // 2.9 ms on a 2060 RTX
 
             // It seems that just compute-rooting all the stencils is
-            // fastest on this GPU, plus some unrolling to share loads
-            // between adjacent pixels.
+            // fastest on this GPU, plus some unrolling and aggressive
+            // staging to share loads between adjacent pixels.
             Var xi, yi, xii, yii;
             stages.push_back(output);
             for (size_t i = 1; i < stages.size(); i++) {
                 Func &s = stages[i];
+                Func prev = stages[i - 1];
                 x = s.args()[0];
                 y = s.args()[1];
                 s.compute_root()
-                    .gpu_tile(x, y, xi, yi, 64, 16)
+                    .gpu_tile(x, y, xi, yi, 30 * 2, 12)
                     .tile(xi, yi, xii, yii, 2, 2)
                     .unroll(xii)
                     .unroll(yii);
+                prev.in()
+                    .compute_at(s, x)
+                    .tile(prev.args()[0], prev.args()[1], xi, yi, 2, 2)
+                    .vectorize(xi)
+                    .unroll(yi)
+                    .gpu_threads(prev.args()[0], prev.args()[1]);
+                prev.in()
```

Comment:
...I have no idea what `.in().in()` implies for scheduling. 

(In general, no pun intended, I have never fully understood the practical use cases for `.in()` and would love to see expanded tutorials on it.)

Reply:
This seems like a good place for some explanatory comments then...

Reply:
PTAL and tell me if the comments clear things up for you

Reply:
Yes, it helps immensely. I had no idea that's how `.in()` could be used. It sounds tremendously useful for situations like this. I'm going to open an issue for "add tutorial lesson for .in()"